main.bal

```diff
@@ -116,13 +100,13 @@ service class Service {
 }
 
 function extractContentLength(readonly & byte[] data) returns int|error {
-    string s = check string:fromBytes(data);
-    int? newLinePos = s.indexOf("\r\n");
+    string firstString = check string:fromBytes(data);
```

Comment:
Here the name `firstString` doesn't make much sense. You may have called this `firstString` because this function is call only for first time, but function is a standalone unit, you should name things within the function thinking only form the scope of the function, ie: forget all the other code, look just at the function when naming local variables in the function.

Also short names are OK for short lived variables.